On Zen4, GCC 13.2.0 control-flow analysis seems to get tripped up when assignment and comparison are combined
inside the if statement.

diff -rupN flux-core-0.74.0.orig/src/common/libutil/basemoji.c flux-core-0.74.0/src/common/libutil/basemoji.c
--- flux-core-0.74.0.orig/src/common/libutil/basemoji.c	2025-05-07 02:46:54.097499500 +0200
+++ flux-core-0.74.0/src/common/libutil/basemoji.c	2025-06-01 16:49:01.088339631 +0200
@@ -147,7 +147,7 @@ int uint64_basemoji_encode (uint64_t id,
 {
     int count;
     int n;
-    char reverse[BASEMOJI_MAXLEN+1];
+    char reverse[BASEMOJI_MAXLEN+1] = {0};
 
     if (buf == NULL || buflen <= 0) {
         errno = EINVAL;
@@ -157,7 +157,8 @@ int uint64_basemoji_encode (uint64_t id,
     /* Encode bytes to emoji (in reverse), which also gives us a count
      * of the total bytes required for this encoding.
      */
-    if ((count = emoji_revenc (reverse, sizeof (reverse), id)) < 0) {
+    count = emoji_revenc (reverse, sizeof (reverse), id);
+    if (count < 0) {
         errno = EINVAL;
         return -1;
     }
